npm install fails because of npm 7

Vaadin 18.0.2, start.vaadin.com fusion project. 

The current pnpm setup doesn't support an easy way of adding npm dependencies (you have to install them with a very specific version of pnpm with `~/.vaadin/node_modules/pnpm/bin/pnpm.js i foo`), so I want to disable pnpm. 

I disabled pnpm in `application.properties` with `vaadin.pnpm.enable=false`

Running the app now fails on npm dependency resolution:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: no-name@undefined
npm ERR! Found: typescript@4.0.3
npm ERR! node_modules/typescript
npm ERR!   dev typescript@"4.0.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer typescript@"^2.7 || ^3" from awesome-typescript-loader@5.2.1
npm ERR! node_modules/awesome-typescript-loader
npm ERR!   dev awesome-typescript-loader@"5.2.1" from the root project
```

I got error
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: no-name@undefined
npm ERR! Found: webpack@5.13.0
npm ERR! node_modules/webpack
npm ERR!   peer webpack@"^4.0.0 || ^5.0.0" from html-webpack-plugin@4.5.1
npm ERR!   node_modules/html-webpack-plugin
npm ERR!     peer html-webpack-plugin@"^3.0.0 || ^4.0.0" from script-ext-html-webpack-plugin@2.1.4
npm ERR!     node_modules/script-ext-html-webpack-plugin
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer webpack@"^1.0.0 || ^2.0.0 || ^3.0.0 || ^4.0.0" from script-ext-html-webpack-plugin@2.1.4
npm ERR! node_modules/script-ext-html-webpack-plugin
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/haijian/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/haijian/.npm/_logs/2021-01-12T12_12_35_479Z-debug.log
```
Two things to check
1. why it tries to use webpack 5
2. Looks like the `script-ext-html-webpack-plugin` is outdated as well. We can see if it's really needed and if we can replace it with sth else.

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
@@ -291,8 +291,7 @@ public abstract class NodeUpdater implements FallibleCommand {
     static Map<String, String> getDefaultDevDependencies() {
         Map<String, String> defaults = new HashMap<>();
 
-        defaults.put("html-webpack-plugin", "3.2.0");
-        defaults.put("script-ext-html-webpack-plugin", "2.1.4");
+        defaults.put("html-webpack-plugin", "4.5.1");
         defaults.put("typescript", "4.0.3");
         defaults.put("ts-loader", "8.0.12");
 

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeUpdater.java
@@ -294,7 +294,7 @@ public abstract class NodeUpdater implements FallibleCommand {
         defaults.put("html-webpack-plugin", "3.2.0");
         defaults.put("script-ext-html-webpack-plugin", "2.1.4");
         defaults.put("typescript", "4.0.3");
-        defaults.put("awesome-typescript-loader", "5.2.1");
+        defaults.put("ts-loader", "8.0.12");
 
         defaults.put("webpack", "4.42.0");
         defaults.put("webpack-cli", "3.3.11");

--- a/flow-server/src/main/resources/webpack.generated.js
+++ b/flow-server/src/main/resources/webpack.generated.js
@@ -201,7 +201,10 @@ module.exports = {
       flowFrontendFolder,
       ...projectStaticAssetsFolders,
     ],
-    extensions: ['.ts', '.js'],
+    extensions: [
+      useClientSideIndexFileForBootstrapping && '.ts',
+      '.js'
+    ].filter(Boolean),
     alias: {
       Frontend: frontendFolder
     }
@@ -230,11 +233,9 @@ module.exports = {
 
   module: {
     rules: [
-      {
+      useClientSideIndexFileForBootstrapping && {
         test: /\.ts$/,
-        use: [
-          'ts-loader'
-        ]
+        loader: 'ts-loader'
       },
       {
         test: /\.css$/i,
@@ -287,7 +288,7 @@ module.exports = {
           }
         }],
       },
-    ]
+    ].filter(Boolean)
   },
   performance: {
     maxEntrypointSize: 2097152, // 2MB

--- a/flow-server/src/main/resources/webpack.generated.js
+++ b/flow-server/src/main/resources/webpack.generated.js
@@ -234,7 +234,7 @@ module.exports = {
       {
         test: /\.ts$/,
         use: [
-          'awesome-typescript-loader'
+          'ts-loader'
         ]
       },
       {

--- a/flow-server/src/main/resources/webpack.generated.js
+++ b/flow-server/src/main/resources/webpack.generated.js
@@ -6,7 +6,6 @@
  */
 const fs = require('fs');
 const HtmlWebpackPlugin = require('html-webpack-plugin');
-const ScriptExtHtmlWebpackPlugin = require('script-ext-html-webpack-plugin');
 const CompressionPlugin = require('compression-webpack-plugin');
 const { InjectManifest } = require('workbox-webpack-plugin');
 const { DefinePlugin } = require('webpack');
@@ -322,10 +321,8 @@ module.exports = {
       template: clientSideIndexHTML,
       filename: indexHtmlPath,
       inject: 'head',
+      scriptLoading: 'defer',
       chunks: ['bundle', ...(devMode ? ['devmodeGizmo'] : [])]
-    }),
-    useClientSideIndexFileForBootstrapping && new ScriptExtHtmlWebpackPlugin({
-      defaultAttribute: 'defer'
     }),
 
     // Service worker for offline

--- a/flow-tests/test-application-theme/test-theme-reusable/src/main/java/com/vaadin/flow/uitest/ui/theme/MyLitField.java
+++ b/flow-tests/test-application-theme/test-theme-reusable/src/main/java/com/vaadin/flow/uitest/ui/theme/MyLitField.java
@@ -23,7 +23,7 @@ import com.vaadin.flow.component.dependency.NpmPackage;
 /**
  * LIT version of vaadin radio button for testing component theming.
  */
-@JsModule("@vaadin/vaadin-radio-button/vaadin-radio-button.ts")
+@JsModule("@vaadin/vaadin-radio-button/vaadin-radio-button.js")
 @Tag("vaadin-radio-button")
 @NpmPackage(value = "@vaadin/vaadin-radio-button", version = "2.0.0-alpha1")
 public class MyLitField extends Component {

--- a/flow-tests/test-embedding/test-embedding-application-theme/src/main/java/com/vaadin/flow/webcomponent/MyLitField.java
+++ b/flow-tests/test-embedding/test-embedding-application-theme/src/main/java/com/vaadin/flow/webcomponent/MyLitField.java
@@ -23,7 +23,7 @@ import com.vaadin.flow.component.dependency.NpmPackage;
 /**
  * LIT version of vaadin radio button for testing component theming.
  */
-@JsModule("@vaadin/vaadin-radio-button/vaadin-radio-button.ts")
+@JsModule("@vaadin/vaadin-radio-button/vaadin-radio-button.js")
 @Tag("vaadin-radio-button")
 @NpmPackage(value = "@vaadin/vaadin-radio-button", version = "2.0.0-alpha1")
 public class MyLitField extends Component {

--- a/flow-tests/test-themes/src/main/java/com/vaadin/flow/uitest/ui/theme/MyLitField.java
+++ b/flow-tests/test-themes/src/main/java/com/vaadin/flow/uitest/ui/theme/MyLitField.java
@@ -23,7 +23,7 @@ import com.vaadin.flow.component.dependency.NpmPackage;
 /**
  * LIT version of vaadin radio button for testing component theming.
  */
-@JsModule("@vaadin/vaadin-radio-button/vaadin-radio-button.ts")
+@JsModule("@vaadin/vaadin-radio-button/vaadin-radio-button.js")
 @Tag("vaadin-radio-button")
 @NpmPackage(value = "@vaadin/vaadin-radio-button", version = "2.0.0-alpha1")
 public class MyLitField extends Component {

